Server crash due to response from one client

The error log is as follows:

```
/root/ethstats-server/lib/history.js:339
                .value();
                 ^

TypeError: _(...).sortBy(...).slice(...).forEach(...).reverse(...).value is not a function
    at History.getNodePropagation (/root/ethstats-server/lib/history.js:339:4)
    at Collection.addBlock (/root/ethstats-server/lib/collection.js:88:46)
    at Sparky.<anonymous> (/root/ethstats-server/app.js:185:10)
    at emitOne (events.js:116:13)
    at Sparky.emit (events.js:211:7)
    at Sparky.incoming (/root/ethstats-server/node_modules/primus-emit/index.js:72:44)
    at transform (/root/ethstats-server/node_modules/primus/spark.js:312:33)
    at Sparky.transforms (/root/ethstats-server/node_modules/primus/spark.js:330:4)
    at decoding (/root/ethstats-server/node_modules/primus/spark.js:215:13)
    at Sparky.decoder (/root/ethstats-server/node_modules/primus/parsers/json.js:34:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ethstats-server@0.2.0 start: `node ./bin/www`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the ethstats-server@0.2.0 start script.
```

Another instance of failure resulted in different logs. This failure was caused when netstats server was restarted and client reconnected itself:

```
2019-03-12 07:45:31.368 [API] [CON] Connected auth03Localhost
/root/ethstats-server/lib/history.js:286
        if( !_.isUndefined(best.height) )
                                ^

TypeError: Cannot read property 'height' of undefined
    at History.bestBlockNumber (/root/ethstats-server/lib/history.js:286:26)
    at Collection.canNodeUpdate (/root/ethstats-server/lib/collection.js:305:55)
    at Collection.requiresUpdate (/root/ethstats-server/lib/collection.js:315:16)
    at Sparky.<anonymous> (/root/ethstats-server/app.js:335:14)
    at emitOne (events.js:116:13)
    at Sparky.emit (events.js:211:7)
    at Sparky.incoming (/root/ethstats-server/node_modules/primus-emit/index.js:72:44)
    at transform (/root/ethstats-server/node_modules/primus/spark.js:312:33)
    at Sparky.transforms (/root/ethstats-server/node_modules/primus/spark.js:330:4)
    at decoding (/root/ethstats-server/node_modules/primus/spark.js:215:13)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ethstats-server@0.2.0 start: `node ./bin/www`
```

Related to #10

--- a/lib/history.js
+++ b/lib/history.js
@@ -276,14 +276,14 @@ History.prototype.prevMaxBlock = function(number)
 
 History.prototype.bestBlock = function()
 {
-	return _.max(this._items, 'height');
+	return _.maxBy(this._items, 'height');
 }
 
 History.prototype.bestBlockNumber = function()
 {
 	var best = this.bestBlock();
 
-	if( !_.isUndefined(best.height) )
+	if( !_.isUndefined(best) && !_.isUndefined(best.height) )
 		return best.height;
 
 	return 0;
@@ -291,14 +291,14 @@ History.prototype.bestBlockNumber = function()
 
 History.prototype.worstBlock = function()
 {
-	return _.min(this._items, 'height');
+	return _.minBy(this._items, 'height');
 }
 
 History.prototype.worstBlockNumber = function(trusted)
 {
 	var worst = this.worstBlock();
 
-	if( !_.isUndefined(worst.height) )
+	if( !_.isUndefined(worst) && !_.isUndefined(worst.height) )
 		return worst.height;
 
 	return 0;
@@ -335,8 +335,7 @@ History.prototype.getNodePropagation = function(id)
 				}
 			}
 		})
-		.reverse()
-		.value();
+		.reverse();
 
 	return propagation;
 }

